configcatclient/rolloutevaluator.py

```diff
@@ -48,174 +49,616 @@ def evaluate(self, key, user, default_value, default_variation_id, settings):  #
+            if str(user_value) not in comparison_value:
+                return True, error
+        # CONTAINS ANY OF
+        elif comparator == Comparator.CONTAINS_ANY_OF:
+            for comparison in comparison_value:
+                if str(comparison) in str(user_value):
+                    return True, error
+        # NOT CONTAINS ANY OF
+        elif comparator == Comparator.NOT_CONTAINS_ANY_OF:
+            if not any(str(comparison) in str(user_value) for comparison in comparison_value):
+                return True, error
+        # IS ONE OF, IS NOT ONE OF (Semantic version)
+        elif Comparator.IS_ONE_OF_SEMVER <= comparator <= Comparator.IS_NOT_ONE_OF_SEMVER:
+            try:
+                match = False
+                for x in filter(None, [x.strip() for x in comparison_value]):
+                    match = semver.VersionInfo.parse(str(user_value).strip()).match('==' + str(x)) or match
+                if (match and comparator == Comparator.IS_ONE_OF_SEMVER) \
+                        or (not match and comparator == Comparator.IS_NOT_ONE_OF_SEMVER):
+                    return True, error
+            except ValueError:
+                validation_error = "'%s' is not a valid semantic version" % str(user_value).strip()
+                error = self._handle_invalid_user_attribute(comparison_attribute, comparator, comparison_value, key,
+                                                            validation_error)
+                return False, error
+        # LESS THAN, LESS THAN OR EQUALS TO, GREATER THAN, GREATER THAN OR EQUALS TO (Semantic version)
+        elif Comparator.LESS_THAN_SEMVER <= comparator <= Comparator.GREATER_THAN_OR_EQUAL_SEMVER:
+            try:
+                if semver.VersionInfo.parse(str(user_value).strip()).match(
+                        COMPARATOR_TEXTS[comparator] + str(comparison_value).strip()
+                ):
+                    return True, error
+            except ValueError:
+                validation_error = "'%s' is not a valid semantic version" % str(user_value).strip()
+                error = self._handle_invalid_user_attribute(comparison_attribute, comparator, comparison_value, key,
+                                                            validation_error)
+                return False, error
+        # =, <>, <, <=, >, >= (number)
+        elif Comparator.EQUALS_NUMBER <= comparator <= Comparator.GREATER_THAN_OR_EQUAL_NUMBER:
+            try:
+                user_value_float = float(str(user_value).replace(",", "."))
+            except ValueError:
+                validation_error = "'%s' is not a valid decimal number" % str(user_value)
+                error = self._handle_invalid_user_attribute(comparison_attribute, comparator, comparison_value, key,
+                                                            validation_error)
+                return False, error
+
+            comparison_value_float = float(str(comparison_value).replace(",", "."))
```

Comment:
Conversion to string looks unnecessary. We should receive a number from a valid config json.